app/routes/analysis/analysis.py

```diff
@@ -118,8 +122,15 @@ async def _zonal_statistics(
     return Response(data=resp["data"])
 
 
-def _get_date_filter(date: str, op: str):
-    if len(date) == 4:
+def _get_date_filter(
+    date: str, op: str, filter_layers: List[RasterLayer]
+) -> Optional[str]:
+    if RasterLayer.umd_tree_cover_loss__year in filter_layers:
+        # only get year for TCL
+        date = date if len(date) == 4 else date[:4]
         return f"umd_tree_cover_loss__year {op} {date}"
+    elif RasterLayer.umd_glad_alerts__date:
+        return f"umd_glad_landsat_alerts__date {op} '{date}'"
     else:
-        return f"umd_glad_landsat_alerts__date {op} {date}"
+        # no date layer to filter by
+        return None
```

Comment:
Good for now.
I believe the date filter should only work on the values of the FROM dataset, right? 
We should work on a more scalable approach in a future release.